pkg/cimfs/cim_writer_windows.go

```diff
@@ -173,21 +234,41 @@ func (c *CimFsWriter) AddLink(oldPath string, newPath string) error {
 	return err
 }
 
-// Unlink deletes the file at `path` from the image.
+// AddMergedLink adds a hard link from `oldPath` to `newPath` in the image. However unlike
+// AddLink this link is resolved at merge time. This allows us to create links to files
+// that are in other CIMs.
+func (c *CimFsWriter) AddMergedLink(oldPath string, newPath string) error {
+	err := c.closeStream()
+	if err != nil {
+		return err
+	}
+	err = winapi.CimCreateMergeLink(c.handle, newPath, oldPath)
+	if err != nil {
+		err = &LinkError{Cim: c.name, Op: "addMergedLink", Old: oldPath, New: newPath, Err: err}
+	}
+	return err
+}
+
+// Unlink deletes the file at `path` from the image. Note that the file MUST have been
+// already added to the image.
 func (c *CimFsWriter) Unlink(path string) error {
 	err := c.closeStream()
 	if err != nil {
 		return err
 	}
-	//TODO(ambarve): CimDeletePath currently returns an error if the file isn't found but we ideally want
-	// to put a tombstone at that path so that when cims are merged it removes that file from the lower
-	// layer
-	err = winapi.CimDeletePath(c.handle, path)
-	if err != nil && !os.IsNotExist(err) {
-		err = &PathError{Cim: c.name, Op: "unlink", Path: path, Err: err}
+	return winapi.CimDeletePath(c.handle, path)
+}
+
+// Adds a tombstone at given path. This ensures that when the the CIMs are merged, the
```

Comment:
There should be a description of what merging CIMs means somewhere. I may have missed it if it's already present though :)

Reply:
I have added a package documentation [here](https://github.com/microsoft/hcsshim/pull/2261/commits/01f38f89bc3e7eed42c1972890e50be37c7e4190#diff-8f37fe68c9649e575646e75dd3eab3a91d10e926f1d71e96e0d332795d5eabc4) , let me know if you want me to add any more details in there.